src/coreclr/jit/abi.cpp

```diff
@@ -206,7 +206,7 @@ ABIPassingInformation::ABIPassingInformation(Compiler* comp, unsigned numSegment
 
     if (numSegments > 1)
     {
-        Segments = new (comp, CMK_ABI) ABIPassingSegment[numSegments];
+        m_segments = new (comp, CMK_ABI) ABIPassingSegment[numSegments];
```

Comment:
`m_segments` and `m_singleSegment` is in a union currently. I didn't check whether it was overall worth it memory wise to separate those fields.. It would increase memory usage for the 1 segment case by a bit while decreasing memory usage in multiple segment cases by a bit more.